test/parallel/test-bootstrap-modules.js

```diff
@@ -3,15 +3,118 @@
+  'NativeModule internal/process/per_thread',
+  'NativeModule internal/process/promises',
+  'NativeModule internal/process/warning',
+  'NativeModule internal/querystring',
+  'NativeModule internal/queue_microtask',
+  'NativeModule internal/timers',
+  'NativeModule internal/url',
+  'NativeModule internal/util',
+  'NativeModule internal/util/debuglog',
+  'NativeModule internal/util/inspect',
+  'NativeModule internal/util/inspector',
+  'NativeModule internal/util/types',
+  'NativeModule internal/validators',
+  'NativeModule path',
+  'NativeModule timers',
+  'NativeModule url',
+  'NativeModule util',
+  'NativeModule vm',
+]);
 
-assert(list.length <= kMaxModuleCount,
-       `Total length: ${list.length}\n` + list.join('\n')
-);
+if (common.isMainThread) {
+  allowedModules.add('NativeModule internal/process/main_thread_only');
+  allowedModules.add('NativeModule internal/process/stdio');
+} else {
+  allowedModules.add('Internal Binding serdes');
+  allowedModules.add('Internal Binding symbols');
+  allowedModules.add('Internal Binding v8');
+  allowedModules.add('Internal Binding worker');
+  allowedModules.add('NativeModule _stream_duplex');
+  allowedModules.add('NativeModule _stream_passthrough');
+  allowedModules.add('NativeModule _stream_readable');
+  allowedModules.add('NativeModule _stream_transform');
+  allowedModules.add('NativeModule _stream_writable');
+  allowedModules.add('NativeModule internal/error-serdes');
+  allowedModules.add('NativeModule internal/process/next_tick');
+  allowedModules.add('NativeModule internal/process/worker_thread_only');
+  allowedModules.add('NativeModule internal/streams/buffer_list');
+  allowedModules.add('NativeModule internal/streams/destroy');
+  allowedModules.add('NativeModule internal/streams/end-of-stream');
+  allowedModules.add('NativeModule internal/streams/legacy');
+  allowedModules.add('NativeModule internal/streams/pipeline');
+  allowedModules.add('NativeModule internal/streams/state');
+  allowedModules.add('NativeModule internal/worker');
+  allowedModules.add('NativeModule internal/worker/io');
+  allowedModules.add('NativeModule module');
+  allowedModules.add('NativeModule stream');
+  allowedModules.add('NativeModule v8');
+  allowedModules.add('NativeModule worker_threads');
+}
+
+if (process.env.NODE_V8_COVERAGE) {
+  allowedModules.add('NativeModule internal/coverage-gen/with_profiler');
+}
+
+const missingModules = new Set([...allowedModules]
+                              .filter((m) => !modulesSet.has(m)));
+const extraModules = new Set([...modulesSet]
+                            .filter((m) => !allowedModules.has(m)));
+const printSet = (s) => { return `{\n${[...s].join(',\n')}\n}\n`; };
+
+assert.deepStrictEqual(modulesSet, allowedModules,
+                       (missingModules.size > 0 ?
+                         `Missing modules ${printSet(missingModules)}` : '') +
+                      `Extra modules ${printSet(extraModules)}`);
```

Comment:
Note to self: Since we have the two set differences we can replace this assert with two asserts (one for each difference set) that assert the difference set is empty.

Reply:
hmm but if I do that then we'll only see the modules in the set for the first assertion to fail.